File: code/Cpp/Day2_MoveAllZerosToEnd.cpp
class Solution {
  public:
    void pushZerosToEnd(vector<int>& arr) {
        
        int posindex=0;
        
        for(int i=0;i<arr.size();i++){
            if(arr[i]!=0){
                arr[posindex]=arr[i];
                posindex++;
            }
        }
        for(int i=posindex;i<arr.size();i++){
            arr[i]=0;
        }
        
    }
};

File: code/Python/Day2_MoveAllZerosToEnd.py
class Solution:
    def pushZerosToEnd(self, arr):
        
        non_zero_index = 0
        
        for i in range(len(arr)):
            if arr[i] != 0:
                arr[non_zero_index] = arr[i]
                non_zero_index += 1
        
        for i in range(non_zero_index, len(arr)):
            arr[i] = 0
